tests/Sniffs/PHP/ForbiddenClassesSniffTest.php

```diff
@@ -0,0 +1,65 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Sniffs\PHP;
+
+use SlevomatCodingStandard\Sniffs\TestCase;
+
+class ForbiddenClassesSniffTest extends TestCase
+{
+
+	public function testNoErrors(): void
+	{
+		self::assertNoSniffErrorInFile(self::checkFile(__DIR__ . '/data/forbiddenClasses.php'));
+	}
+
+	public function testErrors(): void
+	{
+		$report = self::checkFile(__DIR__ . '/data/forbiddenClasses.php', [
+			'forbiddenClasses' => [
+				'\\UserModel' => '\\MyApp\\Models\\User',
+				'\\DB' => '\\TypeORM\\DB',
+				'\\SomeClass' => 'NULL',
+				'\\FullyQualified\\Nested\\SomeClass' => '\\FullyQualified\\SomeClass',
+			],
+			'forbiddenExtends' => [
+				'\\FooNamespace\\ForbiddenExtendedClass' => '\\BarNamespace\\BaseClass',
+			],
+			'forbiddenInterfaces' => [
+				'\\FooNamespace\\SecondImplementedInterface' => '\\BarNamespace\\SecondImplementedInterface',
+				'\\SecondImplementedInterface' => null,
+				'\\ThirdImplementedInterface' => '',
+			],
+			'forbiddenTraits' => [
+				'\\SomeTraitB' => '\\TotallyDifferentTrait',
+				'\\FullyQualified\\SometTotallyDifferentTrait' => '\\DatabaseTrait',
+			],
+		]);
+
+		self::assertSame(19, $report->getErrorCount());
+
+		$allErrors = $report->getErrors();
+
+		self::assertSniffError($report, 9, ForbiddenClassesSniff::CODE_FORBIDDEN_INTERFACE);
+		self::assertCount(2, $allErrors[9]);
+		self::assertSniffError($report, 13, ForbiddenClassesSniff::CODE_FORBIDDEN_TRAIT);
+		self::assertSniffError($report, 14, ForbiddenClassesSniff::CODE_FORBIDDEN_TRAIT);
+		self::assertSniffError($report, 25, ForbiddenClassesSniff::CODE_FORBIDDEN_CLASS);
+		self::assertSniffError($report, 29, ForbiddenClassesSniff::CODE_FORBIDDEN_PARENT_CLASS);
+		self::assertSniffError($report, 29, ForbiddenClassesSniff::CODE_FORBIDDEN_INTERFACE);
+		self::assertSniffError($report, 31, ForbiddenClassesSniff::CODE_FORBIDDEN_TRAIT);
+		self::assertSniffError($report, 38, ForbiddenClassesSniff::CODE_FORBIDDEN_CLASS);
+		self::assertSniffError($report, 40, ForbiddenClassesSniff::CODE_FORBIDDEN_CLASS);
+		self::assertSniffError($report, 43, ForbiddenClassesSniff::CODE_FORBIDDEN_CLASS);
+		self::assertSniffError($report, 47, ForbiddenClassesSniff::CODE_FORBIDDEN_CLASS);
+		self::assertSniffError($report, 48, ForbiddenClassesSniff::CODE_FORBIDDEN_CLASS);
+		self::assertCount(2, $allErrors[48]);
+		self::assertSniffError($report, 49, ForbiddenClassesSniff::CODE_FORBIDDEN_CLASS);
+		self::assertCount(2, $allErrors[49]);
```

Comment:
You should check specific error messages (forth parameter of `assertSniffError()`) - this is useful only on line 48 where both errors are same. The same review is valid for line 9.

Reply:
Changed for message check expect line 48